internal/services/subscription/subscription_locations_data_source.go

```diff
@@ -0,0 +1,113 @@
+package subscription
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2021-01-01/subscriptions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/subscription/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func dataSourceSubscriptionLocations() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceSubscriptionLocationsRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"subscription_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"locations": {
+				Type:     pluginsdk.TypeSet,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"extended_location": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"location": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"type": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceSubscriptionLocationsRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Subscription.Client
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	subscriptionId := d.Get("subscription_id").(string)
+
+	id := parse.NewLocationID(subscriptionId)
+
+	resp, err := client.ListLocations(ctx, id.SubscriptionId, utils.Bool(true))
```

Comment:
what's `true` here?

Reply:
Here `true` means whether the API response includes extended location.